app/jobs/csv_export_job.rb

```diff
@@ -34,28 +34,50 @@ def deploy_csv_export(csv_export)
     filename = csv_export.path_file
     filter = csv_export.filters
 
-    @deploys = Deploy.joins(:stage, :job).where(filter)
-    summary = ["-", "Generated At", csv_export.updated_at, "Deploys", @deploys.count.to_s]
+    deploys = filter_deploys(filter)
+    summary = ["-", "Generated At", csv_export.updated_at, "Deploys", deploys.count.to_s]
     filters_applied = ["-", "Filters", filter.to_json]
 
     CSV.open(filename, 'w+') do |csv|
-      csv << [
-        "Deploy Number", "Project Name", "Deploy Sumary", "Deploy Commit", "Deploy Status", "Deploy Updated",
-        "Deploy Created", "Deployer Name", "Deployer Email", "Buddy Name", "Buddy Email",
-        "Production Flag", "No code deployed"
-      ]
-      @deploys.find_each do |deploy|
-        csv << [
-          deploy.id, deploy.project.name, deploy.summary, deploy.commit, deploy.job.status, deploy.updated_at,
-          deploy.start_time, deploy.job.user.name, deploy.job.user.try(:email), deploy.buddy_name, deploy.buddy_email,
-          deploy.stage.production, deploy.stage.no_code_deployed
-        ]
+      csv << Deploy.csv_header
+      deploys.find_each do |deploy|
+        csv << deploy.csv_line
       end
       csv << summary
       csv << filters_applied
     end
   end
 
+  def filter_deploys(filter)
+    if filter.keys.include?('environments.production')
+      production_value = filter.delete('environments.production')
+      # To match logic of stages.production? True when any deploy_group environment is true or
```

Comment:
best add a comment to `def production?` too ... `# this is reproduced via sql in xxx.rb, make them sync when editing`